src/apps/chahub/tests/test_chahub_mixin.py

```diff
@@ -1,109 +1,137 @@
-#
-#     def test_submission_invalid_not_marked_for_retry_again(self):
-#         # Make submission invalid
-#         self.competition.published = False
-#         self.competition.save()
-#
-#         # Mark submission for retry
-#         submission = CompetitionSubmission(phase=self.phase, participant=self.participant, chahub_needs_retry=True)
-#         # with mock.patch('apps.chahub.models.send_to_chahub') as send_to_chahub_mock:
-#         with mock.patch('chahub.utils.send_to_chahub') as send_to_chahub_mock:
-#             submission.save()
-#             assert not send_to_chahub_mock.called
-#
-#     def test_submission_valid_not_retried_again(self):
-#         # Mark submission for retry
-#         submission = CompetitionSubmission(phase=self.phase, participant=self.participant, chahub_needs_retry=True)
-#         # with mock.patch('apps.chahub.models.send_to_chahub') as send_to_chahub_mock:
-#         with mock.patch('chahub.utils.send_to_chahub') as send_to_chahub_mock:
-#             send_to_chahub_mock.return_value = HttpResponseBase(status=201)
-#             send_to_chahub_mock.return_value.content = ""
-#             submission.save()  # NOTE! not called with force_to_chahub=True as retrying would set
-#             # It does not call send method, only during "do_retries" task should it
-#             assert not send_to_chahub_mock.called
-#
-#     def test_submission_retry_valid_retried_then_sent_and_not_retried_again(self):
-#         # Mark submission for retry
-#         submission = CompetitionSubmission(phase=self.phase, participant=self.participant, chahub_needs_retry=True)
-#         # with mock.patch('apps.chahub.models.send_to_chahub') as send_to_chahub_mock:
-#         with mock.patch('chahub.utils.send_to_chahub') as send_to_chahub_mock:
-#             send_to_chahub_mock.return_value = HttpResponseBase(status=201)
-#             send_to_chahub_mock.return_value.content = ""
-#             submission.save(force_to_chahub=True)
-#             # It does not need retry any more, and was successful
-#             assert send_to_chahub_mock.called
-#             assert not CompetitionSubmission.objects.get(pk=submission.pk).chahub_needs_retry
-#
-#             # reset
-#             send_to_chahub_mock.called = False
-#
-#             # Try sending again
-#             submission.save(force_to_chahub=True)
-#             assert not send_to_chahub_mock.called
+from unittest import mock
+from django.conf import settings
+from django.http.response import HttpResponseBase
+from django.test import TestCase
+
+from competitions.models import Submission
+from factories import UserFactory, CompetitionFactory, DataFactory, SubmissionFactory, PhaseFactory, \
+    CompetitionParticipantFactory
+
+
+class ChaHubTestCase(TestCase):
+    def setUp(self):
+        settings.PYTEST_FORCE_CHAHUB = True
+        # set the url to localhost for tests
+        settings.CHAHUB_API_URL = 'localhost/'
```

Comment:
should include `http(s)`?

Reply:
This variable could be gibberish and it would still work cause I never actually hit the endpoint. I just need _some_ value there. So I figured localhost so that if i had my mock in the wrong place i would just hit myself rather than someone else. 
(quit hitting yourself, quit hitting yourself.... haha)